pkg/provider/beyondtrust/provider_test.go

```diff
@@ -0,0 +1,226 @@
+)
+
+// mock PS API
+func createMockPasswordSafeClient(t *testing.T) kubeclient.Client {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch r.URL.Path {
+
+		case "/Auth/SignAppin":
+			_, err := w.Write([]byte(`{"UserId":1, "EmailAddress":"Felipe"}`))
+			if err != nil {
+				t.Error("Test case Failed")
+			}
+
+		case "/Auth/Signout":
+			_, err := w.Write([]byte(``))
+			if err != nil {
+				t.Error("Test case Failed")
+			}
+
+		case "/secrets-safe/secrets":
+			_, err := w.Write([]byte(`[{"SecretType": "FILE", "Password": "credential_in_sub_3_password","Id": "9152f5b6-07d6-4955-175a-08db047219ce","Title": "credential_in_sub_3"}]`))
+			if err != nil {
+				t.Error("Test case Failed")
+			}
+
+		case "/secrets-safe/secrets/9152f5b6-07d6-4955-175a-08db047219ce/file/download":
+			_, err := w.Write([]byte(`fake_password`))
+			if err != nil {
+				t.Error("Test case Failed")
+			}
+
+		default:
+			http.NotFound(w, r)
+		}
+	}))
+	t.Cleanup(server.Close)
+
+	clientConfig := clientcmd.NewDefaultClientConfig(clientcmdapi.Config{
+		Clusters: map[string]*clientcmdapi.Cluster{
+			"test": {
+				Server: server.URL,
+			},
+		},
+		AuthInfos: map[string]*clientcmdapi.AuthInfo{
+			"test": {
+				Token: "token",
+			},
+		},
+		Contexts: map[string]*clientcmdapi.Context{
+			"test": {
+				Cluster:  "test",
+				AuthInfo: "test",
+			},
+		},
+		CurrentContext: "test",
+	}, &clientcmd.ConfigOverrides{})
+
+	restConfig, err := clientConfig.ClientConfig()
+	assert.Nil(t, err)
+	c, err := kubeclient.New(restConfig, kubeclient.Options{})
+	assert.Nil(t, err)
+
+	return c
+}
+
+func TestNewClient(t *testing.T) {
+	t.Run("should create new client", func(t *testing.T) {
+		ctx := context.Background()
+		p := &Provider{}
+		c := createMockPasswordSafeClient(t)
+		s := esv1beta1.SecretStore{
+			Spec: esv1beta1.SecretStoreSpec{
+				Provider: &esv1beta1.SecretStoreProvider{
+					Beyondtrust: &esv1beta1.BeyondtrustProvider{
+						APIURL:        "https://test.ps-dev.beyondtrustcloud.com:443/BeyondTrust/api/public/v3/",
+						Retrievaltype: "SECRET",
+						Clientid: &esv1beta1.BeyondTrustProviderSecretRef{
+							Value: "12345678-25fg-4b05-9ced-35e7dd5093ae",
+						},
+						Clientsecret: &esv1beta1.BeyondTrustProviderSecretRef{
+							Value: "1234567890on8mTcOzJcltiEg4wYOhDVMer123456789",
+						},
+					},
+				},
+			},
+		}
+
+		_, err := p.NewClient(ctx, &s, c, "test")
+
+		assert.Nil(t, err)
+	})
+
+}
+
+func TestNewClientBadClientId(t *testing.T) {
+	t.Run("should create new client", func(t *testing.T) {
+		ctx := context.Background()
+		p := &Provider{}
+		c := createMockPasswordSafeClient(t)
+		s := esv1beta1.SecretStore{
+			Spec: esv1beta1.SecretStoreSpec{
+				Provider: &esv1beta1.SecretStoreProvider{
+					Beyondtrust: &esv1beta1.BeyondtrustProvider{
+						APIURL:        "https://test.ps-dev.beyondtrustcloud.com:443/BeyondTrust/api/public/v3/",
```

Comment:
please change to APIURL: "[https://example.com:443/BeyondTrust/api/public/v3/](https://example.com/BeyondTrust/api/public/v3/)",